portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,183 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
+/*
+	*Dynamically fetch content from github and render it to html
+
+    *Description of functions below:
+        *getData() asyncronously fetched data from github and calls functions to parse, filter and render component\
+        *SetHeader(repo) fetches and returns certain atributres from repo object
+        *SetRepos(repo) fetches and returns certain atributres from repo object
+        *injectHeader(header) dynamically renders the header to the body
+        *injectRepo(repo) creates a new repository elements and injects it into the body
+
+
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+async function getData (){
+    let response = await fetch('https://api.github.com/users/MohamedShatry/repos');
+    var resJSON = await response.json();
+    let varHeader = SetHeader(resJSON[0].owner);
```

Comment:
you should probably perform a check before accessing the first index?

Reply:
True
Working on that

Reply:
Resolved